source/event_loop.c

```diff
@@ -311,6 +327,50 @@ void aws_event_loop_clean_up_base(struct aws_event_loop *event_loop) {
     aws_hash_table_clean_up(&event_loop->local_data);
 }
 
+void aws_event_loop_register_tick_start(struct aws_event_loop *event_loop) {
+    aws_high_res_clock_get_ticks(&event_loop->latest_tick_start);
+}
+
+void aws_event_loop_register_tick_end(struct aws_event_loop *event_loop) {
+    /* increment the timestamp diff counter (this should always be called from the same thread), the concurrency
+     * work happens during the flush. */
+    uint64_t end_tick = 0;
+    aws_high_res_clock_get_ticks(&end_tick);
+
+    aws_add_size_saturating(event_loop->current_tick_latency_sum, end_tick - event_loop->latest_tick_start);
+    event_loop->latest_tick_start = 0;
+
+    size_t next_flush_time_secs = aws_atomic_load_int(&event_loop->next_flush_time);
+    uint64_t end_tick_secs = aws_timestamp_convert(end_tick, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_SECS, NULL);
+    /* if a second has passed, flush the load-factor. */
+    if (end_tick_secs > next_flush_time_secs) {
+        /* store as seconds because we can't make a 64-bit integer reliably atomic across platforms. */
```

Comment:
trivial: comment about storing as seconds is in a weird place
currently, it's right before we store the tick-latency, which is still in nanoseconds
maybe move it up to where you're setting end_tick_secs